### PR TITLE
Update OAuth registration script

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/jira/servicedesk/projects/SUPPORT/queues/custom/99/SUPPORT-338

### CODE REVIEW

File: utils/oauth-app-registration.sh [hunk 1]
```diff
@@ -61,18 +61,25 @@ OAUTH_APP_ID=$(az ad app list --output tsv --filter "displayName eq '$OAUTH_APP_
...
   )
   echo "Application created, OAUTH_APP_ID=\"$OAUTH_APP_ID\""
+
+  # Get secret
+  OAUTH_APP_SECRET=$(az ad app credential reset \
+    --output tsv \
+    --query "password" \
+    --id "$OAUTH_APP_ID"
+  )
+
```
Comment (michaljurecko) [inline utils/oauth-app-registration.sh:82]: Zmenili sa nazvy niektorych CLI flags + sposob ziskania hesla.

File: utils/oauth-app-registration.sh [hunk 2]
```diff
@@ -102,22 +109,6 @@ if [ ${#perms_arg[@]} -ne 0 ]; then
   fi
 fi
 
-# Set public client = false
-echo "Checking \"publicClient\" property"
-publicClient=$(az ad app list --output tsv --filter "displayName eq '$OAUTH_APP_NAME'"  --query "[].publicClient | [0]")
-if [ "$publicClient" != "false" ]; then
-  echo "Setting publicClient=false"
-  az ad app update --id "$OAUTH_APP_ID"  --set "publicClient=false" || true
-fi
```
Comment (michaljurecko) [inline utils/oauth-app-registration.sh:111]: Toto uz riesi novy flag `is-fallback-public-client`, vid vyssie.

File: utils/oauth-app-registration.sh [hunk 3]
```diff
@@ -102,22 +109,6 @@ if [ ${#perms_arg[@]} -ne 0 ]; then
...
-
-# Allow login with all types of account
-echo "Checking \"signInAudience\" property"
-signInAudience=$(az ad app list --output tsv --filter "displayName eq '$OAUTH_APP_NAME'"  --query "[].signInAudience | [0]")
-if [ "$signInAudience" != "AzureADMultipleOrgs" ]; then
-  echo "WARNING: Property \"signInAudience\" = \"$signInAudience\", but it should by set to \"AzureADMultipleOrgs\"."
-  echo "WARNING: User won't be able to sign in with all types of accounts."
-  echo "WARNING: Please edit it manually in Azure Portal -> App registrations -> $OAUTH_APP_NAME -> Manifest"
-fi
```
Comment (michaljurecko) [inline utils/oauth-app-registration.sh:120]: Toto uz riesi flag `--sign-in-audience AzureADandPersonalMicrosoftAccount `

image

**Comment by michaljurecko** [comment]
Azure CLI zmenilo niektore flags, script uz nefunguje, fixol som to.

CI neprechadza, lebo je tam expirovany OAuth2 token, nemam cas to refreshovat:
- Takze, bud vypnut na chvilu CI check a mergnut to.
- Alebo refreshnut token v CI.

**Review by AdamVyborny — APPROVED** [review]

**Pull request merged**